### ISSUE TITLE
Support directory strategy for remote plugins

### ISSUE THREAD

**Issue by jsok** [bufbuild/buf#1067]
This seems to be purposely disabled:

https://github.com/bufbuild/buf/blob/4e717bd3bfbc5e8b8303d6890d71a476782f41f7/private/buf/bufgen/config.go#L135-L138

This is as much a question as it is a request, because it breaks certain protoc plugins (e.g. [twirp](https://github.com/twitchtv/twirp/blob/ff7d9f87d8598707f3465c80ee5dec1ba8520310/protoc-gen-twirp/generator.go#L181-L216)).

If it's not possible to support, what's the recommended workaround?

## Example

```yaml
version: v1
plugins:
  - remote: buf.build/demolab/plugins/twirp:v8.1.1-1
    out: go
    opt:
      - paths=source_relative
```

**Comment by bufdev** [comment]
Very valid question - let us get you the right response to this. There is reasoning behind it, but want to make sure we get you the full explanation. Stay tuned.

**Comment by mfridman** [comment]
The general consensus was we did this for performance reasons, we could go into more detail, but I'd like to understand the issue you're running in to @jsok, as it may be solvable today.

Since you linked to a very specific part of the Twirp plugin code, I suspect you're running into the following when generating code?

```
error:files have conflicting go_package settings, must be the same: ...
```

If you have a repro I can help take a look, but usually you can workaround this by splitting up your generation steps. I've put together a [repo that highlights the problem](https://github.com/mfridman/buf-generate-twirp-go) using two BSR modules and a potential solution. 

The tl;dr is split up your template into Twirp+Go plugins and Go only. And then generate your Twirp+Go code like normal (without `--include-imports` option) and then generate dependant Go code as a separate buf generate step. Here you'll use the `--template` to target the templates.

**Comment by mfridman** [comment]
Hey @jsok, going to close the issue, but happy to keep the discussion going to figure out if the above workaround is sufficient or there is an actual need to modify the strategy in the Buf CLI.

**Comment by edaniels** [comment]
csharp grpc is broken without this. see: https://github.com/grpc/grpc/issues/28663

**Comment by edaniels** [comment]
@mfridman you may want to reopen

**Comment by mfridman** [comment]
@edaniels thanks for flagging and apologies for the delay in getting back to you.

We recently published a Buf-managed version of the grpc cshrap plugin, available at: https://buf.build/grpc/csharp. Any chance you have a reproducible example of this plugin not behaving correctly?
